src/test/java/com/es/phoneshop/controller/ProductListPageServletTest.java

```diff
@@ -25,17 +24,21 @@
     @Mock
     private RequestDispatcher requestDispatcher;
 
-    private ProductListPageServlet servlet = new ProductListPageServlet();
+    private final ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
-    public void setup(){
+    public void setup() throws ServletException, IOException {
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+        servlet.doGet(request, response);
     }
 
     @Test
-    public void testDoGet() throws ServletException, IOException {
-        servlet.doGet(request, response);
+    public void testDoGetRequestSetAttribute() {
+        verify(request).setAttribute("products", new ArrayListProductDao().findProducts());
```

Comment:
Use mock object instead of real ArrayListProductDao. As described above in Servlet class itself ArrayListProductDao should be moved to a field. The simplest solution is to create a protected setter for this field in servlet class to be used in test.
Feel free to suggest any other solutions though.